src/main/proto/netflix/titus/titus_base.proto

```diff
@@ -336,3 +336,31 @@ message DataRecordMetadata {
     // Time of change.
     uint64 timestamp = 3;
 }
+
+// ----------------------------------------------------------------------------
+// Metadata.
+
+message CollectionMetadata {
+    // Session id defines a scope for revision numbers. Revision numbers are strictly ordered within the same session.
+    // The same entity record may have different revision number across two different sessions.
+    // For example, if session id is associated with a JVM process, and revision numbers are generated again during
+    // bootstrap process, they will change with each process restart.
+    string sessionId = 1;
```

Comment:
How do we know one sessionId has precedence over another? Could sessions include a monotonically increasing `epochId`? Maybe we need some strict ordering on `sessionIds` as well?

Reply:
My assumption is (and I should document it), that there is no ordering requirement on session id. Because of time skew it is impossible to guarantee it in general. This revision numbers make sense within the same session scope, which should be enough to cover the current use cases. We could change it in the future, if we find a way to generate monotonically increasing session id across many machines, not depending on the clock.

Reply:
couldn't we use the leaderId (or leaderEpoch, revisionId) from the underlying leader election provider impl (etcd, zk, etc)?

Reply:
In an offline discussion we agreed on removing revision attribute from this change. We will add it in the future when we have a better strategy for generating globally unique revision numbers.